Documentation/faq.md

```diff
@@ -8,10 +8,50 @@
 
+### Deployment
+
+#### Why odd number of cluster size?
+
+A quorum is the majority of nodes in a cluster. When n is the cluster size, quorum is (n/2)+1. 2 is the quorum both for 3 and 4. The primary goal of etcd is provide reliability and availability. etcd requires the majority (`n/2 + 1` when `n` is the cluster size) to be available to operate. Thus, adding another node to 3-node cluster does not add any value in terms of fault tolerance. And adding 1 node to single-node cluster increases the quorum size and causes leader election; if a wrong node were added by mistake, the cluster would lose its leader and not be able to make any progress. Always start with an odd number of nodes.
+
+#### What is maximum cluster size?
+
+Theoretically, there is no hard limit. However, we recommend no more than 7 nodes. [Google Chubby lock service][chubby], which is a similar project to etcd and widely deployed within Google for many years, suggests running 5 nodes. A 5-member etcd cluster can provide enough fault tolerance in most cases. While larger cluster provides better fault tolerance the write performance reduces since data needs to be replicated to more machines.
+
+#### What is failure tolerance?
+
+etcd cluster continues to operate as long as quorum of cluster can communicate with each other. etcd automatically handles transient failures (e.g. network partitions and delays) while ensuring the safety properties in Raft. etcd can also survive hardware failures (e.g. power outages) by persisting Write-Ahead-Log files onto disk, which contain all stream of events that happened. User just needs to restart the node with same configuration, and etcd will automatically replay its log to the exact point of time before failure.
+
+It is recommended to have an odd number of members in a cluster. Having an odd cluster size doesn't change the number needed for majority, but you gain a higher tolerance for failure by adding the extra member. You can see this in practice when comparing even and odd sized clusters:
+
+| Cluster Size | Majority | Failure Tolerance |
+|:-:|:-:|:-:|
+| 1 | 1 | 0 |
+| 2 | 2 | 0 |
+| 3 | 2 | 1 |
+| 4 | 3 | 1 |
+| 5 | 3 | 2 |
+| 6 | 4 | 2 |
+| 7 | 4 | 3 |
+| 8 | 5 | 3 |
+| 9 | 5 | 4 |
+
+As you can see, adding another member to bring the size of cluster up to an odd size is always worth it. During a network partition, an odd number of members also guarantees that there will almost always be a majority of the cluster that can continue to operate and be the source of truth when the partition ends.
+
 ### Operation
 
 #### How to backup a etcd cluster?
 
 etcdctl provides a `snapshot` command to create backups. See [backup] for more details.
 
-[backup]: https://github.com/coreos/etcd/blob/master/Documentation/op-guide/recovery.md#snapshotting-the-keyspace
\ No newline at end of file
+### Performance
+
+#### How should I benchmark etcd?
+
+Try [benchmark] CLI. Also see [benchmark result][benchmark-result].
```

Comment:
Try the [benchmark] tool. Current [benchmark results][benchmark-result] are available for comparison.